flake.nix

```diff
@@ -0,0 +1,210 @@
+            # add tsort since it's not pulled in for whatever reason
+            buildInputs = prev.buildInputs ++ [ final.tsort ];
+          });
+        };
+        scope' = scope.overrideScope' scopeOverlay;
+
+        # for development
+        devOpamPackages = builtins.attrValues
+          (pkgs.lib.getAttrs (builtins.attrNames devOpamPackagesQuery) scope');
+
+        # osemgrep/semgrep-core
+        # package with all opam deps but nothing else
+        baseOpamPackage = scope'.${package}; # Packages from devPackagesQuery
+
+        osemgrep = baseOpamPackage.overrideAttrs (prev: rec {
+          pname = "osemgrep";
+          buildInputs = prev.buildInputs ++ osemgrepInputs;
+          # all the dune files of semgrep treesitter <LANG>
+          # are missing the :standard field. Basically all compilers autodetct if something
+          # is c or c++ based on file extension, and add the c stdlib based on that. Nix doesn't
+          # because reasons: https://github.com/NixOS/nixpkgs/issues/150655
+          # Dune also passes -xc++ if it detects a c++ file (again sane), but it's included
+          # in the :standard var, which we don't add because ???
+          # TODO add and commit them instead of doing this
+          buildPhase' = ''
+            for f in $(find -type f -iname "dune");do
+              substituteInPlace $f \
+                --replace "flags -fPIC" "flags :standard -fPIC"
+            done
+            make core
+          '';
+          buildPhaseFail = ''
+              echo "Derivation ${pname} won't build outside of a nix shell without submodules:"
+              echo "  nix build '.?submodules=1#' # build from local sources"
+              echo "  nix build '<uri>?submodules=1#' # build from remote sources"
+              echo "  nix run '.?submodules=1#osemgrep' # run osemgrep from local sources"
+              echo "  nix run '<uri>.?submodules=1#osemgrep' # run osemgrep from remote source"
+              exit 1
+          '';
+          # make sure we have submodules
+          # See https://github.com/NixOS/nix/pull/7862
+          buildPhase = if self.submodules then osemgrep.buildPhase' else osemgrep.buildPhaseFail;
+          # TODO check phase
+
+          # DONE! Copy semgrep binaries!!!!
+          installPhase = ''
+            mkdir -p $out/bin
+            cp _build/install/default/bin/* $out/bin
+          '';
+
+        });
+
+
+        # pysemgrep inputs
+
+        devPipInputs = with pythonPackages; [
+          pkgs.git
+          flaky
+          pytest-snapshot
+          pytest-mock
+          pytest-freezegun
+          types-freezegun
+        ];
+
+        # pysemgrep
+        pysemgrep = with pythonPackages; buildPythonApplication {
+          # thanks to @06kellyjac
+          pname = "semgrep";
+          inherit (osemgrep) version;
+          src = ./cli;
+          # TODO checks
+          doCheck = false;
+
+          propagatedBuildInputs = [
```

Comment:
maybe add some coupling: comment in this file and the related pypi file so if
one adds a python dependency, we know we also need to update this file.

Reply:
added